numpy/core/src/_simd/_simd_inc_arg.h

```diff
@@ -0,0 +1,82 @@
+#ifndef _SIMD_SIMD_INC_H_
+    #error "never use it directly, include "_simd_inc.h" instead"
+#endif
+/************************************
+ ** Protected Definitions
+ ************************************/
+static int
+simd_arg_from_obj(PyObject *obj, simd_arg *arg)
```

Comment:
My suggestion is that this become `simd_arg.dispatch.c`, and...
```suggestion
#include "_simd_inc.h" instead"

NPY_NO_EXPORT int
simd_arg_from_obj(PyObject *obj, simd_arg *arg)
```

Reply:
your suggestion requires static dispatching via macros NPY_CPU_DISPATCH_DECLARE and NPY_CPU_DISPATCH_CURFX.

Reply:
I was worried that might be the case. Perhaps for now then:

* add a comment to the top of each file saying something like
  ```c
  /**
   * This file is included by `_simd.dispatch.c.src`. Its contents are affected by the simd configuration, and
   * therefore must be built multiple times. Making it a standalone `.c` file with `NPY_VISIBILITY_HIDDEN`
   * symbols would require judicious use of `NPY_CPU_DISPATCH_DECLARE` and `NPY_CPU_DISPATCH_CURFX`, which was
   * deemed too harmful to readability.
   */
   ```
* change the file extension to `.inc` to make it clear that it's neither a standalone header, nor a standalone `.c` file.
* If (and only if) github doesn't then recognize it as `c` syntax, add a hint to `.gitattributes`.

Reply:
Sounds good to me, thanks!

Reply:
> If (and only if) github doesn't then recognize it as c syntax, add a hint to .gitattributes.

no need to add highlight hint, github already recognize it as `c`